frontend/src/utils/json-parser.ts

```diff
@@ -0,0 +1,465 @@
+            };
+            content: string;
+            lineNumber: number;
+          }[];
+          path: string;
+        }[];
+        location: {
+          domainName: string;
+          location: string;
+          organization: string;
+          repoName: string;
+        };
+        outputFolderName: string;
+        users: {
+          checkedFileTypeContribution: number;
+          commits: {
+            commitResults: CommitResult[];
+            date: string;
+            deletions: number;
+            insertions: number;
+          }[];
+          dailyCommits: {
+            commitResults: CommitResult[];
+            date: string;
+          }[];
+          displayName: string;
+          fileTypeContribution: {
+            [key:string]: number;
+          };
+          location: string;
+          name: string;
+          repoId: string;
+          repoName: string;
+          searchPath: string;
+          variance: number;
+        }[]
+      };
+    };
+    sinceDate: string;
+    untilDate: string;
+    repoSenseVersion: string;
+    isSinceDateProvided: boolean;
+    isUntilDateProvided: boolean;
+    DOMAIN_URL_MAP: {
+      NOT_RECOGNIZED: {
+        BASE_URL: string;
+        BLAME_PATH: string;
+        BRANCH: string;
+        COMMIT_PATH: string;
+        HISTORY_PATH: string;
+        REPO_URL: string;
+      };
+      github: {
+        BASE_URL: string;
+        BLAME_PATH: string;
+        BRANCH: string;
+        COMMIT_PATH: string;
+        HISTORY_PATH: string;
+        REPO_URL: string;
+      },
+    };
+  }
+}
+
+const REPORT_DIR: string = '.';
+const summarySchema = z.object({
+  repoSenseVersion: z.string(),
+  reportGeneratedTime: z.string(),
+  reportGenerationTime: z.string(),
+  zoneId: z.string(),
+  reportTitle: z.string(),
+  repos: z.array(
+      z.object({
+        location: z.object({
+          domainName: z.string(),
+          location: z.string(),
+          repoName: z.string(),
+          organization: z.string(),
+        }),
+        branch: z.string(),
+        displayName: z.string(),
+        outputFolderName: z.string(),
+      }),
+  ),
+  errorSet: z.array(
+      z.object({ repoName: z.string(), errorMessage: z.string() }),
+  ),
+  sinceDate: z.string(),
+  untilDate: z.string(),
+  isSinceDateProvided: z.boolean(),
+  isUntilDateProvided: z.boolean(),
+  supportedDomainUrlMap: z.object({
+    NOT_RECOGNIZED: z.object({
+      BASE_URL: z.string(),
+      BLAME_PATH: z.string(),
+      BRANCH: z.string(),
+      COMMIT_PATH: z.string(),
+      HISTORY_PATH: z.string(),
+      REPO_URL: z.string(),
+    }),
+    github: z.object({
+      BASE_URL: z.string(),
+      BLAME_PATH: z.string(),
+      BRANCH: z.string(),
+      COMMIT_PATH: z.string(),
+      HISTORY_PATH: z.string(),
+      REPO_URL: z.string(),
+    }),
+  }),
+});
+
+const commitsSchema = z.object({
+  authorFileTypeContributionMap: z.object({}).catchall(z.object({}).catchall(z.number())),
```

Comment:
A little confusing, is it possible to use Records or Maps?

Reply:
Thanks for pointing this out. This is possible.